tlair/src/main/scala/at/forsyte/apalache/tla/lir/io/CounterexampleWriter.scala

```diff
@@ -98,55 +101,48 @@ class TlaCounterexampleWriter(writer: PrintWriter) extends CounterexampleWriter
   }
 }
 
-
 class TlcCounterexampleWriter(writer: PrintWriter) extends TlaCounterexampleWriter(writer) {
 
   override def write(rootModule: TlaModule, notInvariant: NotInvariant, states: List[NextState]): Unit = {
-    writer.print(
-      s"""@!@!@STARTMSG 2262:0 @!@!@
-         |Created by Apalache on ${Calendar.getInstance().getTime}
-         |@!@!@ENDMSG 2262 @!@!@
-         |@!@!@STARTMSG 2110:1 @!@!@
-         |Invariant is violated.
-         |@!@!@ENDMSG 2110 @!@!@
-         |@!@!@STARTMSG 2121:1 @!@!@
-         |The behavior up to this point is:
-         |@!@!@ENDMSG 2121 @!@!@
-         |""".stripMargin)
-
-    states.zipWithIndex.foreach {
-      case (state, j) =>
-        val i = j + 1 // start counting from 1, for compatibility with TLC counterexamples
-        val prefix = if(i==1) s"$i: <Initial predicate>" else s"$i: <Next>"
-
-        writer.println(
-          s"""@!@!@STARTMSG 2217:4 @!@!@
-             |$prefix""".stripMargin)
-        printStateFormula(new PrettyWriter(writer), state._2)
-        writer.println(
-          """|
+    writer.print(s"""@!@!@STARTMSG 2262:0 @!@!@
+                    |Created by Apalache on ${Calendar.getInstance().getTime}
+                    |@!@!@ENDMSG 2262 @!@!@
+                    |@!@!@STARTMSG 2110:1 @!@!@
+                    |Invariant is violated.
+                    |@!@!@ENDMSG 2110 @!@!@
+                    |@!@!@STARTMSG 2121:1 @!@!@
+                    |The behavior up to this point is:
+                    |@!@!@ENDMSG 2121 @!@!@
+                    |""".stripMargin)
+
+    states.zipWithIndex.foreach { case (state, j) =>
+      val i = j + 1 // start counting from 1, for compatibility with TLC counterexamples
```

Comment:
There is probably something wrong here, as index 0 would correspond to the constant initializer.

Reply:
Hmm, I didn't modify `TlcCounterexampleWriter` (beyond reformatting), since it wasn't mentioned in the issue, but you're probably right. There are actually quite a few `i +1`s peppered throughout the file, now that I looked at it.